pallets/asset/src/lib.rs

```diff
@@ -657,6 +686,80 @@ decl_module! {
             Ok(())
         }
 
+        /// Creates a checkpoint schedule. Can only be called by the token owner primary or
+        /// secondary key. Only one schedule is allowed for an asset. In order to change an existing
+        /// schedule, it must be removed first.
+        ///
+        /// # Arguments
+        /// * `origin` - A primary or secondary key of the asset owner.
+        /// * `ticker` - The asset ticker.
+        /// * `schedule` - The checkpoint schedule to be assigned to the asset.
+        ///
+        /// # Errors
+        /// * `Unauthorized` - The caller does not own the asset.
+        /// * `CheckpointScheduleAlreadyExists` - A schedule already exists.
+        //
+        // TODO: Make the weight inversely proportional to the length of the period?
+        //
+        #[weight = T::DbWeight::get().reads_writes(6, 2) + 1_000_000_000]
+        pub fn create_checkpoint_schedule(
+            origin,
+            ticker: Ticker,
+            schedule: CheckpointSchedule
+        ) -> DispatchResult {
+            let primary_did = Identity::<T>::ensure_origin_call_permissions(origin)?.primary_did;
+            ensure!(Self::is_owner(&ticker, primary_did), Error::<T>::Unauthorized);
+            ensure!(
+                !<CheckpointSchedules>::contains_key(&ticker, &primary_did),
+                Error::<T>::CheckpointScheduleAlreadyExists
+            );
```

Comment:
Should we allow multiple schedules per ticker? For situations like:
1) Create a checkpoint every 2 years
2) Create a checkpoint every 5 years

If this is not a hard requirement, I'd suggest not implementing :)

cc @adamdossa

Reply:
@maxsam4 In your example I would simply go for a common denominator, e.g., a checkpoint every year. Slightly more checkpoints but less code and storage complexity.

Reply:
For now we are going to have at most one schedule per ticker.